File: docs/code-examples/mesh3d_indexed.py
"""Log a simple colored triangle."""
import rerun as rr

rr.init("rerun_example_mesh3d_indexed", spawn=True)

rr.log(
    "triangle",
    rr.Mesh3D(
        [[0.0, 1.0, 0.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
        vertex_normals=[0.0, 0.0, 1.0],
        vertex_colors=[[0, 0, 255], [0, 255, 0], [255, 0, 0]],
        mesh_properties=rr.cmp.MeshProperties(vertex_indices=[2, 1, 0]),
        mesh_material=rr.cmp.Material(albedo_factor=[0xCC, 0x00, 0xCC, 0xFF]),
    ),
)

File: docs/snippets/all/mesh3d_indexed.rs
//! Log a simple colored triangle with indexed drawing.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_mesh3d_indexed").spawn()?;

    rec.log(
        "triangle",
        &rerun::Mesh3D::new([[0.0, 1.0, 0.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0]])
            .with_vertex_normals([[0.0, 0.0, 1.0]])
            .with_vertex_colors([0x0000FFFF, 0x00FF00FF, 0xFF0000FF])
            .with_mesh_properties(rerun::MeshProperties::from_triangle_indices([[2, 1, 0]])),
    )?;

    Ok(())
}
